internal/alb/lb/attributes.go

```diff
@@ -0,0 +1,204 @@
+	raw, err := c.elbv2.DescribeLoadBalancerAttributes(&elbv2.DescribeLoadBalancerAttributesInput{
+		LoadBalancerArn: aws.String(desired.LbArn),
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to retrieve attributes from ELBV2 in AWS: %s", err.Error())
+	}
+
+	current, _ := NewAttributes(raw.Attributes)
+
+	changeSet, ok := attributesChangeSet(current, desired)
+	if ok {
+		albctx.GetLogger(ctx).Infof("Modifying ELBV2 attributes to %v.", log.Prettify(changeSet))
+		_, err = c.elbv2.ModifyLoadBalancerAttributes(&elbv2.ModifyLoadBalancerAttributesInput{
+			LoadBalancerArn: aws.String(desired.LbArn),
+			Attributes:      changeSet,
+		})
+		if err != nil {
+			eventf, ok := albctx.GetEventf(ctx)
+			if ok {
+				eventf(api.EventTypeWarning, "ERROR", "%s attributes modification failed: %s", desired.LbArn, err.Error())
+			}
+			return fmt.Errorf("failed modifying attributes: %s", err)
+		}
+
+	}
+	return nil
+}
+
+// attributesChangeSet returns a list of elbv2.LoadBalancerAttribute required to change a into b
+func attributesChangeSet(a, b *Attributes) (changeSet []*elbv2.LoadBalancerAttribute, ok bool) {
+	if a.DeletionProtectionEnabled != b.DeletionProtectionEnabled && b.DeletionProtectionEnabled != DeletionProtectionEnabled {
+		changeSet = append(changeSet, &elbv2.LoadBalancerAttribute{
+			Key:   aws.String(DeletionProtectionEnabledString),
+			Value: aws.String(fmt.Sprintf("%v", b.DeletionProtectionEnabled)),
+		})
+	}
+
+	if a.AccessLogsS3Enabled != b.AccessLogsS3Enabled && b.AccessLogsS3Enabled != AccessLogsS3Enabled {
+		changeSet = append(changeSet, &elbv2.LoadBalancerAttribute{
+			Key:   aws.String(AccessLogsS3EnabledString),
+			Value: aws.String(fmt.Sprintf("%v", b.AccessLogsS3Enabled)),
+		})
+	}
+
+	if a.AccessLogsS3Bucket != b.AccessLogsS3Bucket && b.AccessLogsS3Bucket != AccessLogsS3Bucket {
+		changeSet = append(changeSet, &elbv2.LoadBalancerAttribute{
+			Key:   aws.String(AccessLogsS3BucketString),
+			Value: aws.String(b.AccessLogsS3Bucket),
+		})
+	}
+
+	if a.AccessLogsS3Prefix != b.AccessLogsS3Prefix && b.AccessLogsS3Prefix != AccessLogsS3Prefix {
+		changeSet = append(changeSet, &elbv2.LoadBalancerAttribute{
+			Key:   aws.String(AccessLogsS3PrefixString),
+			Value: aws.String(b.AccessLogsS3Prefix),
+		})
+	}
+
+	if a.IdleTimeoutTimeoutSeconds != b.IdleTimeoutTimeoutSeconds && b.IdleTimeoutTimeoutSeconds != IdleTimeoutTimeoutSeconds {
+		changeSet = append(changeSet, &elbv2.LoadBalancerAttribute{
+			Key:   aws.String(IdleTimeoutTimeoutSecondsString),
+			Value: aws.String(fmt.Sprintf("%v", b.IdleTimeoutTimeoutSeconds)),
+		})
+	}
+
+	if a.RoutingHTTP2Enabled != b.RoutingHTTP2Enabled && b.RoutingHTTP2Enabled != RoutingHTTP2Enabled {
+		changeSet = append(changeSet, &elbv2.LoadBalancerAttribute{
+			Key:   aws.String(RoutingHTTP2EnabledString),
+			Value: aws.String(fmt.Sprintf("%v", b.RoutingHTTP2Enabled)),
+		})
+	}
+
+	if len(changeSet) > 0 {
```

Comment:
i think we don't need the ok here, we can test for len(changeSet) that used the result of attributesChangeSet.
which is more clear to express the logic ("if changeSet is not empty, then..")
😸